java/WordCountOptimized.java

```diff
@@ -217,33 +207,66 @@ public static int tokenizeAndSubmitBlock(TObjectIntCustomHashMap<byte[]> counts,
         }
         return 0;
     }
-    
-    private static void submitWord(TObjectIntCustomHashMap<byte[]> m, byte[] word) {
-        m.adjustOrPutValue(word, 1, 1);        
+
+    private static final int TABLE_SIZE = 1024 * 1024;
+    private static final int TABLE_MASK = TABLE_SIZE - 1;
+
+    static class WordMap {
+        final byte[][] keys = new byte[TABLE_SIZE][];
+        final int[] values = new int[TABLE_SIZE];
+        int size = 0;
     }
-    
+
+    private static int submitWord(byte[][] keys, int[] values, int size,
+                                   byte[] word, int off, int len) {
+        int hashCode = (int) XX_HASH.hashBytes(word, off, len);
+        int index = hashCode & TABLE_MASK;
+        while (true) {
+            byte[] key = keys[index];
+            if (key != null) {
+                if (equal(key, word, off, len)) {
+                    values[index]++;
+                    return size;
+                }
+                index = (index + 1) & TABLE_MASK;
+            } else {
+                keys[index] = Arrays.copyOfRange(word, off, off + len);
+                values[index] = 1;
+                return size + 1;
+            }
+        }
+    }
+
+    private static boolean equal(byte[] key, byte[] word, int off, int len) {
```

Comment:
Try this: http://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/   Or MinGW is the older option (Docker is another).